### PR TITLE
all: use updated charmstore dependency

### PR DESCRIPTION
**Pull request by rogpeppe**
We need to update the tests to cater for the new publish
semantics; also we can't use the DELETE endpoint to test
authorization because it's been removed for the time being.

### CODE REVIEW

File: charmstore_test.go [hunk 1]
```diff
@@ -126,9 +125,9 @@ func (s *charmStoreBaseSuite) addCharmNoRevision(c *gc.C, urlStr, name string) (
 	url, err := s.client.UploadCharm(id, ch)
 	c.Assert(err, gc.IsNil)
 
-	// Allow read permissions to everyone.
-	err = s.client.Put("/"+url.Path()+"/meta/perm/read", []string{params.Everyone})
-	c.Assert(err, jc.ErrorIsNil)
+	c.Logf("uploaded %v to %v", id, url)
```
Comment (mhilton) [inline charmstore_test.go]: Is this left over debugging, or generally useful?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/202/
Test PASSed.

File: charmstore_test.go [hunk 2]
```diff
@@ -147,15 +146,27 @@ func (s *charmStoreBaseSuite) addBundle(c *gc.C, urlStr, name string) (charm.Bun
...
-	err = s.client.Put("/"+id.Path()+"/meta/perm/read", []string{params.Everyone})
-	c.Assert(err, jc.ErrorIsNil)
+	s.setPublic(c, id)
 
 	// Return the bundle and its URL.
 	return b, id
 }
 
+func (s *charmStoreBaseSuite) setPublic(c *gc.C, id *charm.URL) {
+
```
Comment (frankban) [inline charmstore_test.go]: Unnecessary newline.

File: charmstore_test.go [hunk 3]
```diff
@@ -147,15 +146,27 @@ func (s *charmStoreBaseSuite) addBundle(c *gc.C, urlStr, name string) (charm.Bun
...
+	// Publish to stable.
+	err := s.client.Put("/"+id.Path()+"/publish?channel=unpublished", &params.PublishRequest{
```
Comment (frankban) [inline charmstore_test.go]: s.client.WithChannel?

File: charmstore_test.go [hunk 4]
```diff
@@ -147,15 +146,27 @@ func (s *charmStoreBaseSuite) addBundle(c *gc.C, urlStr, name string) (charm.Bun
...
+		Channels: []params.Channel{params.StableChannel},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Allow read permissions to everyone.
+	err = s.client.Put("/"+id.Path()+"/meta/perm/read?channel=stable", []string{params.Everyone})
```
Comment (frankban) [inline charmstore_test.go]: Ditto.

**Comment by mhilton** [comment]
:+1:

File: csclient/csclient_test.go [hunk 5]
```diff
@@ -1429,3 +1440,15 @@ func (s *suite) TestWhoAmI(c *gc.C) {
 	c.Assert(err, gc.IsNil)
 	c.Assert(response.User, gc.Equals, "bob")
 }
+
+func (s *suite) setPublic(c *gc.C, id *charm.URL) {
```
Comment (frankban) [inline csclient/csclient_test.go]: Define it on a common embedded test struct?
Or just as a function receiving a client?

**Comment by frankban** [comment]
:+1: with a couple of minor suggestions.

File: csclient/csclient_test.go (hunk 5 above)
Comment (rogpeppe) [inline csclient/csclient_test.go]: It's in a different package (charmrepo vs csclient). I'm not sure I have the energy to make a new shared package for this function right now.

File: charmstore_test.go (hunk 2 above)
Comment (rogpeppe) [inline charmstore_test.go]: done

File: charmstore_test.go (hunk 3 above)
Comment (rogpeppe) [inline charmstore_test.go]: done

File: charmstore_test.go (hunk 4 above)
Comment (rogpeppe) [inline charmstore_test.go]: done

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/203/
Test PASSed.

**Pull request merged**